UserTools/DirectionGridCheck/DirectionGridCheck.cpp

```diff
@@ -0,0 +1,245 @@
+
+  /////////////////// Useful header ///////////////////////
+  if(configfile!="") m_variables.Initialise(configfile); // loading config file
+  //m_variables.Print();
+
+  m_variables.Get("OutputFile", OutputFile);
+  m_variables.Get("ShowEvent", showEvent);
+  m_variables.Get("verbosity", verbosity);
+
+  fOutput_tfile = new TFile("DGCheck.root", "recreate");
+  AngularPlot = new TGraph();
+  AngularPlot->SetTitle("Seed Directions in Angular Space");
+  AngularPlot->GetXaxis()->SetTitle("Polar Angle");
+  AngularPlot->GetYaxis()->SetTitle("ZenithAngle");
+  AngularPlot->SetMarkerStyle(3);
+  AngularPlot->SetMarkerColor(1);
+  AngularPlot->SetLineColor(0);
+  TrueDirPlot = new TGraph();
+  TrueDirPlot->SetTitle("TrueDirection");
+  TrueDirPlot->SetMarkerColor(3);
+  TrueDirPlot->SetMarkerStyle(25);
+  TrueDirPlot->SetLineColor(0);
+  MRDDirPlot = new TGraph();
+  MRDDirPlot->SetTitle("MRDDirection");
+  MRDDirPlot->SetMarkerColor(2);
+  MRDDirPlot->SetMarkerStyle(26);
+  MRDDirPlot->SetLineColor(0);
+  SimpDirPlot = new TGraph();
+  SimpDirPlot->SetTitle("SimpleDirection");
+  SimpDirPlot->SetMarkerColor(4);
+  SimpDirPlot->SetMarkerStyle(27);
+  SimpDirPlot->SetLineColor(0);
+  RecoDirPlot = new TGraph();
+  RecoDirPlot->SetMarkerColor(5);
+  RecoDirPlot->SetMarkerStyle(28);
+  RecoDirPlot->SetLineColor(0);
+
+  m_data= &data; //assigning transient data pointer
+  /////////////////////////////////////////////////////////////////
+
+  return true;
+}
+
+
+bool DirectionGridCheck::Execute(){
+
+    auto* annie_event = m_data->Stores["RecoEvent"];
+    if (!annie_event) {
+        Log("Error: The PhaseITreeMaker tool could not find the ANNIEEvent Store",
+            0, verbosity);
+        return false;
+    }
+
+    // MC entry number
+    m_data->Stores.at("ANNIEEvent")->Get("MCEventNum", fMCEventNum);
+
+    // MC trigger number
+    m_data->Stores.at("ANNIEEvent")->Get("MCTriggernum", fMCTriggerNum);
+
+    // ANNIE Event number
+    m_data->Stores.at("ANNIEEvent")->Get("EventNumber", fEventNumber);
+
+    auto get_flags = m_data->Stores.at("RecoEvent")->Get("EventFlagged", fEventStatusFlagged);
+
+    if ((fEventStatusFlagged) != 0) {
+        //  if (!fEventCutStatus){
+        Log("PhaseIITreeMaker Tool: Event was flagged with one of the active cuts.", v_debug, verbosity);
+        return true;
+    }
+
+    // Only check this event
+    //if (showEvent > 0 && (int)fEventNumber != showEvent) return true;
+    if (eventCount > 9)return true;
+
+    VertexGeometry* myvtxgeo = VertexGeometry::Instance();
+    Direction tempdir;
+
+
+    if (eventCount == 0) {
+        for (int l = 0; l < 100; l++) {
+            double phi = (6 * TMath::Pi() / 50) * l;
+            double theta = (TMath::Pi() / 200) * l;
+
+            while (phi > 2 * TMath::Pi()) {
+                std::cout << "correcting theta" << endl;
+                phi -= 2 * TMath::Pi();
+            }
+
+            tempdir.SetTheta(theta);
+            tempdir.SetPhi(phi);
```

Comment:
tempdir does not seem to be used anywhere. It gets declared, its value set 100 times, and never used.